<launch>
    <node pkg="hku_m100_landing" type="hku_m100_landing_preprocessor" name="hku_m100_preprocessor_node" >
    </node>

      <node pkg="hku_m100_landing" type="hku_m100_landing_controller" name="hku_m100_controller_node" >
    </node>

    <node ns="hku_m100_landing" name="velocity_x_controller" pkg="pid" type="controller" output="screen" >
      <param name="node_name" value="apriltag_tracking_x_controller" />
      <param name="Kp" value="-0.3" />
      <param name="Ki" value="-0.02" />
      <param name="Kd" value="-0.02" />
      <param name="upper_limit" value="5" />
      <param name="lower_limit" value="-5" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/setpoint" />
      <param name="topic_from_plant" value="/teamhku/tag_x" />
      <param name="topic_from_controller" value="/teamhku/velocity_control_effort_x" />
     </node>

    <node ns="hku_m100_landing" name="velocity_y_controller" pkg="pid" type="controller" output="screen" >
      <param name="node_name" value="apriltag_tracking_y_controller" />
      <param name="Kp" value="-0.3" />
      <param name="Ki" value="-0.02" />
      <param name="Kd" value="-0.02" />
      <param name="upper_limit" value="5" />
      <param name="lower_limit" value="-5" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/setpoint" />
      <param name="topic_from_plant" value="/teamhku/tag_y" />
      <param name="topic_from_controller" value="/teamhku/velocity_control_effort_y" />
    
     </node>

    <node name="setpoint_node" pkg="pid" type="setpoint_node" output="screen" />
<!-- 
   <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
      <rosparam command="delete" param="/diagnostic_aggregator" />
      <rosparam command="load" file="$(find pid)/config/pid_diags.yaml" />
   </node>

    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/teamhku/velocity_control_effort_x /teamhku/tag_x /teamhku/velocity_control_effort_y /teamhku/tag_y /setpoint/data" /> -->


</launch>
